docs/building.md

```diff
@@ -0,0 +1,125 @@
+# Akri Build
+Building Akri, whether locally or in the automated CI builds, leverages the same set of `make` commands.
+
+In essence, there are 3 types of builds for Akri:
+1. Core components 
+    1. Rust code: containers based on Rust code are built using `Cargo cross` and subsequent `docker build` commands include the cross-built binaries.
+        > Note: For Rust code, build/Dockerfile.* does NOT run `cargo build`, instead they simply copy cross-built binaries into the container
+    2. Other code: these containers can be .NET or python or whatever else ... the build/Dockerfile.* must do whatever building is required.
+2. Intermediate builds: these containers are used as part of the build process and are not used in production explicitly
+
+## Core components
+The core Akri components are the containers that are downloaded at runtime to provide Akri's functionality.  They include the agent, the controller, the webhook,  the brokers, and the applications.  The majority of Akri component code is written in Rust, but there are several components that are written in .NET or python.
+
+All of the core components are built with a `make` command.  These are the supporting makefiles:
+* `Makefile`: this provides a single point of entry to build any Akri component
+* `build/akri-containers.mk`: this provides the build and push functionality for Akri containers
+* `build/akri-rust-containers.mk`: this provides a simple definition to build and push Akri components written in Rust
+* `build/akri-dotnet-containers.mk`: this provides a simple definition to build and push Akri components written in .NET
+* `build/akri-python-containers.mk`: this provides a simple definition to build and push Akri components written in Python
+
+### Configurability
+The makefiles allow for several configurations:
+* BUILD_AMD64: if set not to 1, the make commands will ignore AMD64
+* BUILD_ARM32: if set not to 1, the make commands will ignore ARM32
+* BUILD_ARM64: if set not to 1, the make commands will ignore ARM64
+* REGISTRY: allows configuration of the container registry (defaults to imaginary: devcaptest.azurecr.io)
```

Comment:
What is the value to specify if I want to do everything locally only (no remote registry)?

Reply:
i think this is the answer (see e2e py test):

you can specify anything you want for PREFIX as long as you set pullPolicy=Never and pull it from the same PREFIX.

```
make install-cross
make akri-cross-build-<platform>
make <component>-build-<platform>
PREFX="whateveryouwant" LABEL_PREFIX="local" make <component>-build-<platform>
```

then, you just need helm to reference `whateveryouwant/<component>:local-<platform>`

Reply:
Should we mentioned that in the local build example, so folks do not assume they have to create ghcr repo?

Reply:
i added instructions, largely pulled from our test case ... hopefully it works :)